### PR TITLE
Us14

### PR DESCRIPTION
**Pull request by markertmer**
toggle button for status on admin merchants index.

changes the order after clicking button, may have to fix later.

### CODE REVIEW

File: app/controllers/admin/merchants_controller.rb [hunk 1]
```diff
@@ -27,10 +27,15 @@ def create
 
   def update
     merchant = Merchant.find(params[:id])
-
+# binding.pry
     if merchant.update(merchant_params)
-      redirect_to "/admin/merchants/#{merchant.id}"
-      flash[:alert] = "Merchant successfully updated!"
+      if merchant_params.count == 1 # as in, only having :status
```
Comment (mcharrod) [inline app/controllers/admin/merchants_controller.rb:32]: Hey Mark, good job with the conditionals. I will say, the way this conditional is written, I could see the count being equal to 1 on multiple cases. For instance, updating the merchant's name. Please update this before we merge this into developer.

File: app/views/admin/merchants/index.html.erb [hunk 2]
```diff
@@ -4,9 +4,15 @@
 
 <br>
 <% @merchants.each do |merchant| %>
-<li><%= merchant.name %> - status: <%= merchant.status %></li>
+  <li id="merchant-<%= merchant.id %>"><%= merchant.name %> - status: <%= merchant.status %>
+  <%= form_with url: "/admin/merchants/#{merchant.id}", method: :patch, local: true do |form| %>
+  <% statusing = [ "enabled", "enable" ] %>
+  <% statusing = [ "disabled", "disable" ] if merchant.status == "enabled" %>
```
Comment (mcharrod) [inline app/views/admin/merchants/index.html.erb:10]: I like that this button is dynamic! Great job on making it say "enable" or "disable" based on the params.

**Review by mcharrod — CHANGES_REQUESTED** [review]
Hey Mark, I've added some comments. Please make some updates to your Admin::MerchantsController before we merge this into developer. Thank you!

File: app/views/admin/merchants/index.html.erb (hunk 2 above)
Comment (markertmer) [reply app/views/admin/merchants/index.html.erb:10]: Thanks Katy!

File: app/controllers/admin/merchants_controller.rb (hunk 1 above)
Comment (markertmer) [reply app/controllers/admin/merchants_controller.rb:32]: Yeah, I know it's not the best! I'll work on it...

File: app/controllers/admin/merchants_controller.rb [hunk 3]
```diff
@@ -42,8 +47,8 @@ def update
     def merchant_params
       param_hash = {}
 
-      param_hash[:name] = params[:name] unless params[:name] == ""
-      param_hash[:status] = params[:status] unless params[:status] == ""
+      param_hash[:name] = params[:name] unless params[:name] == "" || params[:name] == nil
```
Comment (mcharrod) [inline app/controllers/admin/merchants_controller.rb:50]: Hey Mark, I like that you are trying to guard against blank names. Luckily, our model validations does that for us. It looks like when trying to use this functionality in local host, some things slip by. Have a look at the merchant items controller, and give that a go instead. Let me know if you have questions!

**Review by mcharrod — CHANGES_REQUESTED** [review]
Hey Mark, I've added some comments for some changes that should be made before merging into developer. Looks like with our current set up when we type in a blank name, it redirects without making any changes. Ideally, we should get a flash notification that gives us an error, something our model validations will take care of for us. Have a look at the merchant items controller to see this in action!

**Comment by jwgalvin** [comment]
Pulled down and passes tests.  Nice work implementing the changes requested.  Merging to main.

**Pull request merged**